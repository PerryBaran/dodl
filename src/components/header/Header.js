import React, { useContext } from 'react';
import AppContext from '../AppContext';
import style from './header.module.css';

const Header = (props) => {
    const { hideWhilePlaying } = useContext(AppContext);

    return (
        <header className={`centerFlex ${style.headerContainer}`}>
          <h1 className={`${style.header} ${hideWhilePlaying(style.pause)}`}>Dreaming of Detuned Love</h1>
        </header>
    );
}

export default Header;