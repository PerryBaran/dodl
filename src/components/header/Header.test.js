import React from "react";
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom'
import Header from "./Header";
import AppContext from "../../utils/context/AppContext";
import renderer from "react-test-renderer";

afterEach(cleanup);

const renderHeader = (input) => {
    return (
        <AppContext.Provider value={false}>
            <Header text={input}/>
        </AppContext.Provider>
    )
}

it("renders without crashing", () => { 
    render(renderHeader())
})

it("renders text passed as prop", () => {
    render(renderHeader('biscuit'));
    expect(screen.getByText(/biscuit/)).toBeInTheDocument();
})

it("matches snapshot", () => {
    const tree = renderer.create(renderHeader('biscuit')).toJSON();
    expect(tree).toMatchSnapshot();
})
