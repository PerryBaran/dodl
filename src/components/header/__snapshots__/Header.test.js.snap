// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="centerFlex headerContainer"
>
  <h1
    className="header pause"
  >
    biscuit
  </h1>
</header>
`;
